test/conftest.py

```diff
@@ -226,15 +225,17 @@ def delete_image(runtime, image_name):
 def podman_ee_tag(request):
     image_name = gen_image_name(request)
     yield image_name
-    delete_image('podman', image_name)
+    # FIXME: defer to end?
+    # delete_image('podman', image_name)
 
 
 @pytest.fixture
 @pytest.mark.test_all_runtimes
 def ee_tag(request, runtime):
     image_name = gen_image_name(request)
     yield image_name
-    delete_image(runtime, image_name)
+    # FIXME: defer to end?
+    # delete_image(runtime, image_name)
```

Comment:
Yeah, this fixme definitely needs to be resolved- the problem is that doing the image deletions inline can break the build cache, as IIUC at least some versions of podman will try to clear dangling intermediate layers as part of the deletion, but when multiple builds are occurring in parallel, it can nuke entries that are being used.

I *think* the right way to fix this will be to defer the deletions to the session teardown. IIRC there's only one session under pytest-xdist- if not, we'll have to figure something else out...

And yeah, the fact that it ran faster when you re-enabled this is because one of the longest-running tests failed early, likely because of what I mentioned above. We can definitely clean them up, but doing them inline will almost definitely introduce instability as the test concurrency increases.

Reply:
Gotcha. We gotta do what we gotta do then. We can noodle another way to do the cleanup at a later point.

Reply:
Made a new issue to address in #519